orchestrator/models.py

```diff
@@ -1,28 +1,60 @@
 import uuid
 from enum import IntEnum
 from django.db import models
+from django.utils import timezone
 from django.contrib.postgres.fields import JSONField
 
 
 def message_default():
     message_default_dict = {
         'log': '',
         'failed_jobs': {},
-        'unknown_jobs':{},
+        'unknown_jobs': {},
         'info': ''
     }
     return message_default_dict
 
 
 class Status(IntEnum):
     CREATED = 0
-    PENDING = 1
-    RUNNING = 2
-    COMPLETED = 3
-    FAILED = 4
-    ABORTED = 5
-    UNKNOWN = 6
-    SUSPENDED = 7
+    SUBMITTING = 1
+    SUBMITTED = 2
+    PENDING = 3
+    RUNNING = 4
+    COMPLETED = 5
+    FAILED = 6
+    ABORTED = 7
+    UNKNOWN = 8
+    SUSPENDED = 9
+
+    def transition(self, transition_to):
+        if self == self.CREATED:
+            if transition_to in (self.SUBMITTING, self.ABORTED,):
+                return True
+        elif self == self.SUBMITTING:
+            if transition_to in (self.SUBMITTED, self.ABORTED):
+                return True
+        elif self == self.SUBMITTED:
+            if transition_to in (
+            self.PENDING, self.RUNNING, self.COMPLETED, self.FAILED, self.ABORTED, self.SUSPENDED, self.UNKNOWN):
+                return True
+        elif self == self.PENDING:
+            if transition_to in (self.PENDING, self.RUNNING, self.COMPLETED, self.FAILED, self.ABORTED, self.SUSPENDED, self.UNKNOWN):
+                return True
+        elif self == self.RUNNING:
+            if transition_to in (self.RUNNING, self.COMPLETED, self.FAILED, self.ABORTED, self.SUSPENDED, self.UNKNOWN):
+                return True
+        elif self in (self.COMPLETED, self.FAILED, self.ABORTED,):
+            # Terminal state
+            return False
+        elif self == self.UNKNOWN:
+            # Can transition to all lsf states
+            if transition_to in (self.PENDING, self.RUNNING, self.COMPLETED, self.FAILED, self.ABORTED, self.SUSPENDED, self.UNKNOWN):
+                return True
+        elif self == self.SUSPENDED:
+            if transition_to in (self.PENDING, self.RUNNING, self.ABORTED,):
+                return True
+        return False
```

Comment:
We should add an error message here -- like "Attempting to change status from `COMPLETED` to `RUNNING` for JOB ..."

Reply:
I think for the logger we'll want a Job ID but that can come later